backend/fms_core/viewsets/_utils.py

```diff
@@ -366,69 +384,41 @@ def default_prefilling(sample_id: int, template, user_prefill_data):
         extra_sheet_list = [sheet["name"] for sheet in template["sheets info"] if sheet.get("batch", None) is None]
         # Dictionary to map a sheet to its stitch column
         dict_stitch = {sheet["name"]: sheet.get("stitch_column", None) for sheet in template["sheets info"]}
-        step_dict = {}
-        batch_container_dict = {}
-        for sample_id, step_id in queryset.values_list("sample", "step").distinct():
-            new_step = False
-            new_batch_container = False
-            # without placement there is only 1 destination for each sample
-            if placement_data is None or placement_data.get(str(sample_id)) is None:
-                sample_row_dict = default_prefilling(sample_id, template, user_prefill_data)
-                batch_row_dict = {}
-                # Use step to extract specifications and attach it to the correct sheet and column
-                step = Step.objects.get(id=step_id)
+        # Process case where we have a single batch sheet template without a matching sample sheet
+        if dict_batch_sheet.get(False, None) is None and dict_batch_sheet.get(True, None) is not None and dict_stitch[dict_batch_sheet[True]] is None:
+            sample_id_list, step_id_list = zip(*queryset.values_list("sample", "step"))
```

Comment:
I'm actually surprised `zip` can be used like that too because `zip` normally does the opposite. It takes N different sequences and produces a single sequence of each of which has N items.

Reply:
yeah it's quite a handy little function